### PR TITLE
tag router

### PR DESCRIPTION
**Pull request by sanxun0325**
**What this PR does**: 
support tag router

dubbo tag router:https://dubbo.apache.org/zh/docs/advanced/routing-rule/

not support runtime: https://github.com/apache/dubbo/issues/9713

conditional route will then submit another PR
**Which issue(s) this PR fixes**: 
Fixes #

**You should pay attention to items below to ensure your pr passes our ci test**
We do not merge pr with ci tests failed

- [x] All ut passed (run 'go test ./...' in project root)
- [x] After go-fmt ed , run 'go fmt project' using goland.
- [x] Golangci-lint passed, run 'sudo golangci-lint run' in project root.
- [x] After import formatted, (using [imports-formatter](https://github.com/dubbogo/tools#5-how-to-get-imports-formatter) to run 'imports-formatter .' in project root, to format your import blocks, mentioned in [CONTRIBUTING.md](https://github.com/apache/dubbo-go/blob/master/CONTRIBUTING.md) above) 
- [x] Your new-created file needs to have [apache license](https://raw.githubusercontent.com/dubbogo/resources/master/tools/license/license.txt) at the top, like other existed file does.
- [x] All integration test passed. You can run integration test locally (with docker env). Clone our [dubbo-go-samples](https://github.com/apache/dubbo-go-samples) project and replace the go.mod to your dubbo-go, and run 'sudo sh start_integration_test.sh' at root of samples project root. (M1 Slice is not Support)

### CODE REVIEW

**Comment by sanxun0325** [comment]
Don't merge yet, I need to refine the code

File: cluster/router/tag/factory.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+package tag
+
+import (
+	"dubbo.apache.org/dubbo-go/v3/cluster/router"
+	"dubbo.apache.org/dubbo-go/v3/common/extension"
+)
+
+func init() {
+	extension.SetRouterFactory("tag", NewTagRouterFactory)
```
Comment (justxuewei) [inline cluster/router/tag/factory.go:26]: Please add a const in the `common/constant` package.

Comment (sanxun0325) [reply cluster/router/tag/factory.go]: done

File: cluster/router/chain/chain.go [hunk 2]
```diff
@@ -77,6 +77,9 @@ func (c *RouterChain) AddRouters(routers []router.PriorityRouter) {
 func (c *RouterChain) SetInvokers(invokers []protocol.Invoker) {
 	c.mutex.Lock()
 	c.invokers = invokers
+	for _, v := range c.routers {
+		v.Notify(c.invokers)
+	}
 	c.mutex.Unlock()
```
Comment (AlexStocks) [inline cluster/router/chain/chain.go:83]: 为了安全起见，还是把这里顺便改改，用 defer unlock 吧

Comment (sanxun0325) [reply cluster/router/chain/chain.go]: done

**Review by AlexStocks — APPROVED** [review]

**Review by Mulavar — APPROVED** [review]

**Review by LaurenceLiZhixin — APPROVED** [review]

**Pull request merged**